pkg/kubelet/pod_workers.go

```diff
@@ -138,57 +135,61 @@ type podWorkers struct {
 func newPodWorkers(syncPodFn syncPodFnType, recorder record.EventRecorder, workQueue queue.WorkQueue,
 	resyncInterval, backOffPeriod time.Duration, podCache kubecontainer.Cache) *podWorkers {
 	return &podWorkers{
-		podUpdates:                map[types.UID]chan UpdatePodOptions{},
-		isWorking:                 map[types.UID]bool{},
-		lastUndeliveredWorkUpdate: map[types.UID]UpdatePodOptions{},
-		syncPodFn:                 syncPodFn,
-		recorder:                  recorder,
-		workQueue:                 workQueue,
-		resyncInterval:            resyncInterval,
-		backOffPeriod:             backOffPeriod,
-		podCache:                  podCache,
+		podUpdates:     map[types.UID]chan UpdatePodOptions{},
+		isWorking:      map[types.UID]bool{},
+		syncPodFn:      syncPodFn,
+		recorder:       recorder,
+		workQueue:      workQueue,
+		resyncInterval: resyncInterval,
+		backOffPeriod:  backOffPeriod,
+		podCache:       podCache,
 	}
 }
 
-func (p *podWorkers) managePodLoop(podUpdates <-chan UpdatePodOptions) {
+func (p *podWorkers) managePodLoop(uid types.UID, podUpdates <-chan UpdatePodOptions) {
```

Comment:
What's the point of adding `uid` as an argument here if it just comes through on the update already?

Reply:
If I don't add `uid` as an argument here, when running the unit test, kubelet will receive a signal from `syncCh`, `update.Pod.UID` raise a panic because `update.Pod` is nil. Besides the `managePodLoop` is `uid` specific, so I think using `uid` in an argument to replace getting from `update`.

Reply:
You could also just check for `nil`, which is probably the safer approach. It's best to avoid redundant code where possible.